Tools/autotest/common.py

```diff
@@ -1879,9 +1905,81 @@ def test_arm_feature(self):
+                    self.arm_vehicle()
+                    if not self.disarm_vehicle():
+                        self.progress("Failed to DISARM")
+                        raise NotAchievedException()
+                    self.progress("PASS arm mode : %s" % mode)
+                if mode in self.NOT_ARMABLE_MODES_LIST:
+                    if mode in self.NOT_SETTABLE_MODES_LIST:
+                        self.progress("Not settable mode : %s" % mode)
+                        try:
+                            self.do_set_mode_via_command_long(mode, timeout=2)
+                        except AutoTestTimeoutException:
+                            self.progress("PASS not arm mode : %s" % mode)
+                            pass
+                        except ValueError:
+                            self.progress("PASS not arm mode : %s" % mode)
+                            pass
+                    else:
+                        self.progress("Not armable mode : %s" % mode)
+                        self.do_set_mode_via_command_long(mode)
+                        self.run_cmd(mavutil.mavlink.MAV_CMD_COMPONENT_ARM_DISARM,
+                                     1,  # ARM
+                                     0,
+                                     0,
+                                     0,
+                                     0,
+                                     0,
+                                     0,
+                                     want_result=mavutil.mavlink.MAV_RESULT_FAILED
+                                     )
+                    self.mav.motors_disarmed_wait()
+                    self.progress("PASS not arm mode : %s" % mode)
+                if mode in self.GPS_ARMABLE_MODES_LIST:
+                    self.progress("Armable mode needing GPS : %s" % mode)
+                    self.do_set_mode_via_command_long(mode)
+                    self.arm_vehicle(timeout=30)
+                    if not self.disarm_vehicle():
+                        self.progress("Failed to DISARM")
+                        raise NotAchievedException()
+                    self.progress("PASS arm mode : %s" % mode)
+                    self.progress("Not armable mode without GPS : %s" % mode)
+                    self.wait_gps_disable()
+                    self.do_set_mode_via_command_long(mode)
+                    self.run_cmd(mavutil.mavlink.MAV_CMD_COMPONENT_ARM_DISARM,
+                                 1,  # ARM
+                                 0,
+                                 0,
+                                 0,
+                                 0,
+                                 0,
+                                 0,
+                                 want_result=mavutil.mavlink.MAV_RESULT_FAILED
+                                 )
+                    self.mav.motors_disarmed_wait()
+                    self.set_parameter("SIM_GPS_DISABLE", 0)
+                    self.wait_ekf_happy()
```

Comment:
This wait should really be before trying to arm the vehicle with GPS above.

Reply:
@khancyr You marked this resolved but it's exactly in the same place?

Reply:
ha, I miss this one. But I really  prefer to keep it here, as it ensure that GPS check pass for all other tests... even if it shouldn't matter

Reply:
done